### PR TITLE
ADEN-6631 Adds call for skin in UAP:Roadblock

### PR DESCRIPTION
**Pull request by leshniak**
https://wikia-inc.atlassian.net/browse/ADEN-6631

### CODE REVIEW

File: extensions/wikia/AdEngine/js/template/roadblock.js [hunk 1]
```diff
@@ -6,39 +6,36 @@ define('ext.wikia.adEngine.template.roadblock', [
...
-		if (!context.opts.disableSra) {
-			medrecSlotElement.style.opacity = '0';
-			gptHelper.refreshSlot(medrecSlot.name);
-			medrecSlot.pre('renderEnded', function () {
-				medrecSlotElement.style.opacity = '';
-			});
-			log(['handleMedrec', 'refreshing slot', medrecSlot], log.levels.info, logGroup);
-		}
+	function handleSkin(skinSlotElement) {
+		var skinSlot = slotRegistry.get(skinSlotElement.id);
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/template/roadblock.js]: I believe you can check in `slotRegistry` is slot is present or not instead of checking in DOM

Comment (kvas-damian) [reply extensions/wikia/AdEngine/js/template/roadblock.js]: Also BtfBlocker won't crash if you unblock not existing slot. Maybe we could just unblock it (without any checks) ;)

File: extensions/wikia/AdEngine/js/provider/directGpt.js [hunk 2]
```diff
@@ -61,8 +61,11 @@ define('ext.wikia.adEngine.provider.directGpt', [
 			isPageFairRecoverable: pageFair ? pageFair.isSlotRecoverable : false,
 			sraEnabled: sraEnabled,
 			atfSlots: atfSlots,
-			getAdUnitBuilder: function () {
-				return context.opts.megaAdUnitBuilderEnabled ? megaAdUnitBuilder : kiloAdUnitBuilder;
+			megaAdUnitSlots: ['INVISIBLE_SKIN'],
+			getAdUnitBuilder: function (slotName) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/directGpt.js]: Can we unittest it somehow?
Also please remember we serve `INVISIBLE_SKIN` in Remnant provider

Comment (leshniak) [reply extensions/wikia/AdEngine/js/provider/directGpt.js]: Done.

File: extensions/wikia/AdEngine/js/provider/directGpt.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ define('ext.wikia.adEngine.provider.directGpt', [
 			isPageFairRecoverable: pageFair ? pageFair.isSlotRecoverable : false,
 			sraEnabled: sraEnabled,
 			atfSlots: atfSlots,
-			getAdUnitBuilder: function () {
+			getAdUnitBuilder: function (slotName) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/directGpt.js]: `slotName` is unused here

Comment (leshniak) [reply extensions/wikia/AdEngine/js/provider/directGpt.js]: Fixed

File: extensions/wikia/AdEngine/js/provider/remnantGpt.js [hunk 4]
```diff
@@ -27,6 +28,10 @@ define('ext.wikia.adEngine.provider.remnantGpt', [
 		{
 			afterSuccess: function (slotName) {
 				slotTweaker.removeDefaultHeight(slotName);
+				if (!uapContext.isBfaaLoaded()) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/provider/remnantGpt.js]: Something went wrong with resolving conflicts. This code should be here.

Comment (leshniak) [reply extensions/wikia/AdEngine/js/provider/remnantGpt.js]: done

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**